client/extensions/woocommerce/state/sites/settings/email/actions.js

```diff
@@ -0,0 +1,361 @@
+ *
+ * @param  {Number|String} siteId        Jetpack site ID
+ * @param  {String}        apiKey        MailChimp api key
+ * @return {Function}                    Action thunk
+ */
+export const submitMailChimpApiKey = ( siteId, apiKey ) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	dispatch( mailchimpApiKeySubmit( siteId ) );
+
+	return request( siteId ).put( 'mailchimp/api_key', { mailchimp_api_key: apiKey } )
+		.then( settings => {
+			dispatch( mailchimpApiKeySubmitSuccess( siteId, settings ) );
+		} )
+		.catch( error => {
+			dispatch( mailchimpApiKeySubmitFailure( siteId, error ) );
+		} );
+};
+
+/**
+ * Triggers a network request to set MailChimp store info in MailChimp plugin settings.
+ *
+ * @param  {Number|String} siteId        Jetpack site ID
+ * @param  {Object}        storeInfo     MailChimp store info settings object
+ * @return {Function}                    Action thunk
+ */
+export const submitMailchimpStoreInfo = ( siteId, storeInfo ) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	dispatch( mailchimpStoreInfoSubmit( siteId ) );
+
+	return request( siteId ).put( 'mailchimp/store_info', storeInfo )
+		.then( settings => {
+			dispatch( mailchimpStoreInfoSubmitSuccess( siteId, settings ) );
+		} )
+		.catch( error => {
+			dispatch( mailchimpStoreInfoSubbmitFailure( siteId, error ) );
+		} );
+};
+
+/**
+ * Triggers a network request to set MailChimp campaign defualts
+ * info in MailChimp plugin settings.
+ *
+ * @param  {Number|String} siteId           Jetpack site ID
+ * @param  {Object}        campaignDefaults MailChimp campaign defaults settings object
+ * @return {Function}                       Action thunk
+ */
+export const submitMailchimpCampaignDefaults = ( siteId, campaignDefaults ) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	dispatch( mailchimpCampaignDefaultsSubmit( siteId ) );
+
+	return request( siteId ).put( 'mailchimp/campaign_defaults', campaignDefaults )
+		.then( settings => {
+			dispatch( mailchimpCampaignDefaultsSubmitSuccess( siteId, settings ) );
+		} )
+		.catch( error => {
+			dispatch( mailchimpCampaignDefaultsSubmitFailure( siteId, error ) );
+		} );
+};
+
+/**
+ * Triggers a network request to fetch current mailing list available created for account
+ * asociated with api key storred in plugin config..
+ *
+ * @param  {Number|String} siteId        Jetpack site ID
+ * @return {Function}                    Action thunk
+ */
+export const requestLists = ( siteId ) => ( dispatch, getState ) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	dispatch( mailchimpListsRequest( siteId ) );
+
+	return request( siteId ).get( 'mailchimp/newsletter_setting' )
```

Comment:
Side note - at first I thought this might have been an incorrect endpoint name since we are requesting "lists" and the endpoint name is singular `newsletter_setting`

Reply:
uhh right, wording could be a bit better here. Will keep that in mind for later when there will be time for not critical cleanup.